api/types/resource.go

```diff
@@ -321,3 +360,10 @@ Outer:
 
 	return true
 }
+
+func compareStrByDir(a string, b string, isDesc bool) bool {
```

Comment:
nit: Something about this name seems funny. I don't know if it's because I keep translating `Dir` to `Directory` in my head or because `StrByDir` rhymes and is funny to say. A few suggestions but you are welcome to ignore or use an alternative.

```suggestion
func stringCompare(a string, b string, isDesc bool) bool {
```

```suggestion
func lexicographicalLess(a string, b string, isDesc bool) bool {
```

Reply:
went with `stringCompare`